src/lsliced.jl

```diff
@@ -0,0 +1,109 @@
+struct SlicePartial{Syms1,P}
+    __p::P
+end
+
+@inline function Base.getproperty(p::SlicePartial{Val{s1}},s2::Symbol) where s1
+  getfield(p, :__p)[Val(s1), Val(s2)]
+end
+
+@inline function Base.setproperty!(p::SlicePartial{Val{s1}},s2::Symbol,y) where s1
+  getfield(p, :__p)[Val(s1), Val(s2)] = y
+end
+
+@inline function Base.getproperty(x::Sliced,s::Symbol)
+  if s == :__x
+    return getfield(x,:__x)
+  end
+  SlicePartial{Val{s},typeof(x)}(x)
+end
+
+
+#####################################
+# Array Interface 
+# (getindex/getproperty methods are shared with SLArray
+#####################################
+Base.propertynames(::LArray{T,N,S}) where {T,N,S<:Tuple{Syms1,Syms2}} where {Syms1,Syms2} = Syms1, Syms2
+symnames(::Type{LArray{T,N,S}}) where {T,N,S<:Tuple{Syms1,Syms2}} where {Syms1,Syms2} = Syms1, Syms2
+
+@inline Base.getindex(x::Sliced,i::Int...) = getfield(x,:__x)[i...]
+@inline Base.getindex(x::Sliced,s1::Symbol,s2::Symbol) = getindex(x,Val(s1),Val(s2))
+@inline Base.getindex(x::Sliced,i1::Int,s2::Symbol) = getindex(x,i1::Int,Val(s2))
+@inline Base.getindex(x::Sliced,s1::Symbol,i2)   = getindex(x,Val(s1),i2)
+
+@inline @generated function Base.getindex(x::Sliced,::Val{s1},::Val{s2}) where {s1,s2}
+  idx1 = findfirst(y->y==s1,symnames(x)[1])
+  idx2 = findfirst(y->y==s2,symnames(x)[2])
+  :(getfield(x,:__x)[$idx1, $idx2])
+end
+
+@inline @generated function Base.getindex(x::Sliced,i1::Int,::Val{s2}) where s2
+  idx2 = findfirst(y->y==s2,symnames(x)[2])
+  :(getfield(x,:__x)[i1, $idx2])
+end
+
+@inline @generated function Base.getindex(x::Sliced,::Val{s1},i2::Int) where s1
+  idx1 = findfirst(y->y==s1,symnames(x)[1])
+  :(getfield(x,:__x)[$idx1, i2])
+end
+
+
+
+@inline Base.setindex!(x::LSliced,y,i...) = getfield(x,:__x)[i...] = y
+@inline Base.setindex!(x::LSliced,y,s1::Symbol,s2::Symbol) = setindex!(x,y,Val(s1),Val(s2))
+@inline Base.setindex!(x::LSliced,y,s1::Symbol,i2) = setindex!(x,y,Val(s1),i2)
+@inline Base.setindex!(x::LSliced,y,i1,s2::Symbol) = setindex!(x,y,i1,Val(s2))
+@inline Base.setindex!(x::LSliced,y,s1::Symbol,i2::Val) = setindex!(x,y,Val(s1),i2)
+@inline Base.setindex!(x::LSliced,y,i1::Val,s2::Symbol) = setindex!(x,y,i1,Val(s2))
+
+@inline @generated function Base.setindex!(x::LSliced,y,::Val{s1},::Val{s2}) where {s1, s2}
+    idx1 = findfirst(y->y==s1,symnames(x)[1])
+    idx2 = findfirst(y->y==s2,symnames(x)[2])
+    :(x.__x[$idx1, $idx2] = y)
+end
+
+@inline @generated function Base.setindex!(x::LSliced,y,::Val{s1},i2::Number) where s1
+    idx1 = findfirst(y->y==s1,symnames(x)[1])
+    :(x.__x[$idx1, i2] = y)
+end
+
+@inline @generated function Base.setindex!(x::LSliced,y,i1::Number,::Val{s2}) where s2
+    idx2 = findfirst(y->y==s2,symnames(x)[2])
+    :(x.__x[i1, $idx2] = y)
+end
+
+
+#####################################
+# Broadcast
+#####################################
+LAStyle{T,N,Syms}(x::Val{2}) where {T,N,Syms<:Tuple} = LAStyle{T,N,Syms}()
```

Comment:
Why `Val{2}`?

Reply:
Honestly because the tests break without that...

I'm not really across how ArrayStyle works but the compiler complained about that line not existing when testing broadcast. It did make me wonder if that was also required for regular two dimensional LArrays and just wasn't tested yet.